module/actor/data/character-components.ts

```diff
@@ -0,0 +1,169 @@
+    formula: new fields.StringField({ required: true, nullable: false }),
+    damagetype: new fields.StringField({ required: true, nullable: false }),
+    extdamagetype: new fields.StringField({ required: true, nullable: false }),
+    hitlocation: new fields.StringField({ required: true, nullable: false }),
+    accumulate: new fields.BooleanField({ required: true, nullable: true, initial: false }),
+    count: new fields.NumberField({ required: false, nullable: false, initial: undefined }),
+    roll: new fields.StringField({ required: true, nullable: true, initial: '' }),
+  }
+}
+
+type DamageActionSchema = ReturnType<typeof damageActionSchema>
+
+/* ---------------------------------------- */
+
+const modifierSetSchema = () => {
+  return {
+    modifiers: new fields.ArrayField(new fields.StringField({ required: true, nullable: false }), {
+      required: true,
+      nullable: false,
+    }),
+  }
+}
+
+/* ---------------------------------------- */
+
+const conditionsSchema = () => {
+  return {
+    actions: new fields.SchemaField(
+      {
+        maxActions: new fields.NumberField({ required: true, nullable: false, initial: 1 }),
+        maxBlocks: new fields.NumberField({ required: true, nullable: false, initial: 1 }),
+      },
+      { required: true, nullable: false }
+    ),
+    posture: new fields.StringField({ required: true, nullable: false }),
+    // NOTE: change from previous schema where maneuver is not nullable
+    maneuver: new fields.StringField({ required: true, nullable: true }),
+    move: new fields.StringField({ required: true, nullable: false }),
+
+    self: new fields.SchemaField(modifierSetSchema(), { required: true, nullable: false }),
+    target: new fields.SchemaField(modifierSetSchema(), { required: true, nullable: false }),
+    // Change from previous schema. Now a Set instead of an Array to eliminate duplicates.
+    usermods: new fields.SetField(new fields.StringField({ required: true, nullable: false }), {
+      required: true,
+      nullable: false,
+    }),
+
+    reeling: new fields.BooleanField({ required: true, nullable: false, initial: false }),
+    exhausted: new fields.BooleanField({ required: true, nullable: false, initial: false }),
+
+    damageAccumulators: new fields.ArrayField(
+      new fields.SchemaField(damageActionSchema(), { required: true, nullable: false }),
+      { required: true, nullable: false }
+    ),
+  }
+}
+
+type ConditionsSchema = ReturnType<typeof conditionsSchema>
+
+/* ---------------------------------------- */
+
+const reactionSchema = () => {
+  return {
+    // NOTE: change from previous schema, where "modifier" was a string
+    modifier: new fields.NumberField({ required: true, nullable: false, initial: 0 }),
+    situation: new fields.StringField({ required: true, nullable: false }),
+    modifierTags: new fields.StringField({ required: true, nullable: false }),
+  }
+}
+
+type ReactionSchema = ReturnType<typeof reactionSchema>
```

Comment:
These need a better name -- it is used for both ReactionModifiers and ConditionalModifiers.